cloudcix_primitives/storage_kvm.py

```diff
@@ -55,38 +59,55 @@ def build(
     # Define message
     messages = {
         1000: f'1000: Successfully created storage {storage}',
-        3021: f'3021: Failed to connect to the host {host}',
-        3022: f'3022: Failed to create storage_kvm {domain_path}{storage} on the host {host}'
+        3001: f'3001: Already a storage {storage} of different size than requested size {size}GBs '
+              f'file exists at {domain_path}',
+        3021: f'3021: Failed to connect to the host {host} for the payload read_storage_file',
+        3022: f'3022: Failed to connect to the host {host} for the payload create_storage_file',
+        3023: f'3023: Failed to create storage_kvm {domain_path}{storage} on the host {host}',
     }
 
-    message_list = {}
-
-    # Define payload
-    payload = f'qemu-img create -f qcow2 {domain_path}{storage} {size}G'
-
-    # Create storage using SSH communication
-    response = comms_ssh(
-        host_ip=host,
-        payload=payload,
-        username='robot',
-    )
-    if response['channel_code'] != CHANNEL_SUCCESS:
-        msg = f'{messages[3021]}\n ' \
-                f'channel_code: {response["channel_code"]}s.\n' \
-                f'channel_message: {response["channel_message"]}\n' \
-                f'channel_error: {response["channel_error"]}'
-        message_list.append(msg)
-        success = False
-    if response['payload_code'] != SUCCESS_CODE:
-        msg = f'{messages[3022]}\n ' \
-                f'payload_code: {response["payload_code"]}s.\n' \
-                f'payload_message: {response["payload_message"]}\n' \
-                f'payload_error: {response["payload_error"]}'
-        message_list.append(msg)
-        success = False
-
-    message_list.append(messages[1000])
-    return success, message_list
+    def run_host(host, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, host, 'robot')
+        fmt = HostErrorFormatter(
+            host,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        payloads = {
+            'read_storage_file': f'qemu-img info {domain_path}{storage}',
+            'create_storage_file': f'qemu-img create -f qcow2 {domain_path}{storage} {size}G',
+        }
+
+        ret = rcc.run(payloads['read_storage_file'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, messages[prefix + 1]), fmt.successful_payloads
+        create_storage = True
+        if ret["payload_code"] == SUCCESS_CODE:
+            # No need to create storage drive exists already
+            create_storage = False
+            # storage already exists, extract the size of the file
+            size_match = re.search(r'virtual size: (\S+)', ret["payload_message"].strip())
+            checked_size = size_match.group(1) if size_match else 0
+            if int(checked_size) != int(size):
+                return True, messages[3001], fmt.successful_payloads
```

Comment:
```suggestion
                return False, messages[3001], fmt.successful_payloads
```
Something has gone wrong and an error should be raised. We do not want to fail silently. 

If we are not notified, we will not know that build was called twice for different sizes in the request and there is a problem.

Reply:
Sorry my bad